lib/circuitry/cli.rb

```diff
@@ -0,0 +1,66 @@
+require 'circuitry/provisioner'
+require 'thor'
+
+module Circuitry
+  class CLI < Thor
+    class_option :verbose, aliases: :v, type: :boolean
+
+    desc 'provision <queue> -t <topic> [<topic> ...]', 'Provision a queue subscribed to one or more topics'
+
+    long_desc <<-END
+      Creates an SQS queue and dead letter queue named <queue>[-failures] with appropriate
+      SNS access policy along with one or more SNS topics named <topic> that has an SQS subscription for
+      each.
+
+      When the queue already exists, its policy will be added or updated.
+
+      When a topic already exists, it will be ignored.
+
+      When a topic subscription already exists, it will be ignored.
+    END
+
+    option :topics, aliases: :t, type: :array, required: :true
+    option :access_key, aliases: :a
+    option :secret_key, aliases: :s
+    option :region, aliases: :r
+
+    def provision(queue_name)
+      with_custom_config(queue_name) do |config|
+        logger = Logger.new(STDOUT)
+        logger.level = Logger::INFO if options['verbose']
+        Circuitry::Provisioner.new(config, logger).run
```

Comment:
I think `logger` should be a hash option instead of a dedicated argument.

```ruby
Circuitry::Provisioner.new(config, logger: logger).run

def initialize(config, logger: Logger.new(STDOUT)
  # ...
end
```